metrics-core/src/main/java/software/amazon/swage/metrics/record/cloudwatch/CloudWatchRecorder.java

```diff
@@ -99,6 +99,14 @@
                 .build();
     }
 
+    private final Runnable sendAggregatedDataRunnable = () -> {
```

Comment:
This runnable seems excessive/unnecessary.  Probably should catch and log or swallow the exception(s) lower down.  In the sendData method perhaps.

For robustness I could see having some error handling logic around the actual sending, perhaps retrying if CloudWatch is being flaky.

Reply:
Actually, I just remembered my reasoning here: I want to ensure that ANY exception or error is caught, no matter how the code changes in the future. If I wrapped this around `metricsClient.putMetricData( request )`, for example, someone could write code that is part of the runnable and throws, but we wouldn't catch and log it, leading to the exact problem we have today.

Reply:
How about [handling Throwables in the Executor](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html#afterExecute-java.lang.Runnable-java.lang.Throwable-)?

Given there's a TODO to allow the Executor to be provided, then that would enable the behaviour to be configured there rather always being logged.

Reply:
I'm not sure that allowing an Executor to be injected adds any value for customers, since as a customer, I don't really care how my metrics get to CloudWatch, only that they get there. I don't think that having visibility into what is going wrong and why my metrics aren't showing up in CloudWatch is a special case that should require customers to inject something to handle it themselves.